include/h2o/socketpool.h

```diff
@@ -200,6 +201,10 @@ void h2o_socketpool_detach(h2o_socketpool_t *pool, h2o_socket_t *sock);
  * determines if a socket belongs to the socket pool
  */
 static int h2o_socketpool_is_owned_socket(h2o_socketpool_t *pool, h2o_socket_t *sock);
+/**
+ * sets the address family to be used when connecting to targets
+ */
+static void h2o_socketpool_set_address_family(h2o_socketpool_t *pool, int address_family);
```

Comment:
Considering that other variables inside the `h2o_socketpool_t` are accessed directly (modulo the ones that start with `_`), maybe we do not need to define a setter.